packages/genesys-spark-components/src/components/beta/gux-avatar-group/gux-avatar-group.tsx

```diff
@@ -0,0 +1,203 @@
+import {
+  Component,
+  h,
+  JSX,
+  Element,
+  State,
+  Prop
+  // writeTask
+} from '@stencil/core';
+import { trackComponent } from '@utils/tracking/usage';
+import { GuxAvatarAccent, GuxAvatarSize } from '../gux-avatar/gux-avatar.types';
+
+interface ProcessedAvatar {
+  focusableAvatar: HTMLGuxAvatarFocusableBetaElement;
+  avatar: HTMLGuxAvatarBetaElement;
+  button?: HTMLButtonElement;
+  link?: HTMLAnchorElement;
+  img?: HTMLImageElement;
+  name: string;
+  accent: GuxAvatarAccent;
+  isOverflow: boolean;
+}
+
+/**
+ * @slot - Some gux-avatars
+ */
+
+@Component({
+  styleUrl: 'gux-avatar-group.scss',
+  tag: 'gux-avatar-group-beta',
+  shadow: true
+})
+export class GuxAvatarGroup {
+  @Element()
+  root: HTMLElement;
+
+  private listItemClickListeners: Map<HTMLElement, (event: Event) => void> =
+    new Map();
+
+  @Prop()
+  avatarLimit: number = 7;
+
+  @State()
+  avatarList: HTMLGuxAvatarFocusableBetaElement[];
+
+  @State()
+  private processedAvatars: ProcessedAvatar[] = [];
+
+  get countAvatars() {
+    return this.avatarList.length ?? 0;
+  }
+
+  async componentWillLoad(): Promise<void> {
+    this.processAvatars();
+
+    trackComponent(this.root);
+  }
+
+  disconnectedCallback() {
+    if (this.listItemClickListeners.size > 0) {
+      this.removeAllListeners();
+    }
+  }
+
+  private getAccent(index): string {
+    return `${(index % 12) + 1}` as GuxAvatarAccent;
+  }
+
+  private processAvatars(): void {
+    const avatarElements = Array.from(
+      this.root.children
+    ) as HTMLGuxAvatarFocusableBetaElement[];
+
+    this.processedAvatars = avatarElements.map((focusableAvatar, index) => {
+      const avatar = focusableAvatar.querySelector(
+        'gux-avatar-beta'
+      ) as HTMLGuxAvatarBetaElement;
+      const button = focusableAvatar.querySelector('button');
+      const link = focusableAvatar.querySelector('a');
+      const img = avatar.querySelector('img');
+      return {
+        focusableAvatar,
+        avatar,
+        button,
+        link,
+        img,
+        name: avatar.name,
```

Comment:
You will probably want optional chaining operators on these if the values are null eg `name : avatar?.name  || ''` or where u are doing the query selection.